backend/dataall/modules/s3_datasets_shares/services/share_object_service.py

```diff
@@ -537,73 +538,6 @@ def _validate_group_membership(session, share_object_group, environment_uri):
                 message=f'Team: {share_object_group} is not a member of the environment {environment_uri}',
             )
 
-    @staticmethod
```

Comment:
This is moved to the share_item_service because now it applies to each item, not to the whole share